e2e-tests/src/mock_server.rs

```diff
@@ -0,0 +1,47 @@
+use httpmock::MockServer;
+use serde_json::Value;
+
+/// MockServerGlobal (has mock server inside)
+pub struct MockServerGlobal {
+    pub(crate) client_url: String,
+    pub(crate) mock_server: MockServer,
+}
+
+impl MockServerGlobal {
+    /// To create a new client
+    pub fn new() -> Self {
+        let server = MockServer::start();
+        Self { client_url: format!("http://localhost:{:?}", server.port()), mock_server: server }
+    }
+
+    /// To get mutable mock server ref for adding expects for URLs
+    pub fn mut_mock_server(&mut self) -> &mut MockServer {
+        &mut self.mock_server
+    }
+
+    /// To get the server URL
+    pub fn url(&self) -> String {
+        self.client_url.clone()
+    }
+
+    /// To add mock on the mock server endpoints
+    pub fn add_mock_on_endpoint(
+        &mut self,
+        path: &str,
+        body_contains: Option<&str>,
+        body_contains_2: Option<&str>,
```

Comment:
doesn't look like the correct design, what are we trying to do here

Reply:
Yeah this code is not clear. So basically we have two. conditions to be inserted in URL so that's why I made this function need to replace this with correct implementation.